includes/Checker/Abstract_Check_Runner.php

```diff
@@ -627,7 +627,7 @@ final protected function get_slug() {
 	private function get_check_context() {
 		$plugin_basename = $this->get_plugin_basename();
 		$plugin_path     = is_dir( $plugin_basename ) ? $plugin_basename : WP_PLUGIN_DIR . '/' . $plugin_basename;
-		return new Check_Context( $plugin_path, $this->get_slug() );
+		return new Check_Context( $plugin_path, $this->get_slug() ? $this->get_slug() : basename( $plugin_path, '.php' ) );
```

Comment:
I think this should happen within `Plugin_Context` (which `Check_Context` extends).

That's where there's already logic to determine the slug by default, if none was specified. It doesn't make sense to me to have that duplicated in this class, neither here nor in the `set_slug()` method where it was so far.

I think it makes most sense to have `get_slug()` pass through the empty string that `get_slug_param()` may return (for AJAX always, for CLI only if no slug was explicitly specified). And then because that string is empty, `Plugin_Context` would define the default slug as the source of truth.

Alternatively, if there's value to have the logic here in `Abstract_Check_Runner`, we should remove the logic from `Plugin_Context` and make the constructor `$slug` parameter required. It just doesn't make sense to have the logic split/duplicated between two places.

Reply:
I just moved this there in 16f8e0998a9d882481464709148dab7380b750c3 because otherwise it didn't work when using WP-CLI "shortinit" with `--require`. We don't pass enough information to `Check_Context`/`Plugin_Context` on its own to determine whether it's a single-file plugin or not.

Feel free to make edits to the PR with what you have in mind.


----

Aside: whenever we have to add workarounds like this I'm wondering whether we actually really want to support single-file plugins or not. They aren't allowed in the repo anyway.

Reply:
How does WP-CLI "shortinit" work? Not familiar with it... How does it prevent that logic in the constructor from working correctly?

Reply:
Thanks for clarifying. I can have a look later this week to see if I can come up with a solution for that.